File: MPP-lab-project/src/Domain/DTOs/RaceDTO.cs
namespace MPP_lab_project.Domain.DTOs;

public class RaceDTO
{
    public SwimmingDistances SwimmingDistance { get; }
    public SwimmingStyles SwimmingStyle { get; }
    public int NoSwimmers { get; }

    public RaceDTO(SwimmingDistances swimmingDistance, SwimmingStyles swimmingStyle, int noSwimmers)
    {
        SwimmingDistance = swimmingDistance;
        SwimmingStyle = swimmingStyle;
        NoSwimmers = noSwimmers;
    }
}

File: swimming-contest-software/Model/src/main/java/swimmingApp/domain/dtos/RaceDTO.java
package swimmingApp.domain.dtos;

import swimmingApp.domain.enums.SwimmingDistances;
import swimmingApp.domain.enums.SwimmingStyles;

public class RaceDTO {
    private final SwimmingDistances swimmingDistance;
    private final SwimmingStyles swimmingStyle;
    private final Integer noSwimmers;

    public RaceDTO(SwimmingDistances swimmingDistance, SwimmingStyles swimmingStyle, Integer noSwimmers) {
        this.swimmingDistance = swimmingDistance;
        this.swimmingStyle = swimmingStyle;
        this.noSwimmers = noSwimmers;
    }

    public SwimmingDistances getDistance() {
        return swimmingDistance;
    }

    public SwimmingStyles getStyle() {
        return swimmingStyle;
    }

    public Integer getNoSwimmers() {
        return noSwimmers;
    }
}
